paasta_tools/kubernetes_tools.py

```diff
@@ -1855,20 +1853,26 @@ def sanitize_for_config_hash(
         :param config: complete_config hash to sanitise
         :returns: sanitised copy of complete_config hash
         """
-        ahash = {
-            key: copy.deepcopy(value)
-            for key, value in config.to_dict().items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
-        spec = ahash["spec"]
-        ahash["spec"] = {
-            key: copy.deepcopy(value)
-            for key, value in spec.items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
+        ahash = config.to_dict()  # deep convert to dict
         ahash["paasta_secrets"] = get_kubernetes_secret_hashes(
             service=self.get_service(), environment_variables=self.get_env()
         )
+
+        # remove data we dont want used to hash configs
+        # replica count
+        if ahash.get("spec", {}).get("replicas"):
+            del ahash["spec"]["replicas"]
+        # soa-configs SHA
+        try:
+            for container in ahash["spec"]["template"]["spec"]["containers"]:
```

Comment:
should we do the same type of `get()` that we do above here? if it was possible for `spec` to not exist in the previous conditional, it’s also possible here, right?

Reply:
I re-checked and it isn't possible for the keys to not exist because they are converted from k8s objects, but it is possible for the values to be `None` if they aren't explicitly. I've removed the `KeyError` exception, but left the `TypeError` catch just in case.

Reply:
I wonder if it'd be too noisy to add a log line when we hit this case